src/api/server.ts

```diff
@@ -2,39 +2,52 @@
 
 import { fetch as tauriFetch } from '@tauri-apps/api/http';
 import { Child, ChildProcess, Command as TauriCommand } from '@tauri-apps/api/shell';
+import React from 'react';
+
+import { noop } from '../lib/noop';
 
 export const REFSTUDIO_HOST = 'http://localhost:1487';
 
 interface StartingState {
   state: 'starting';
 }
-interface ReadyState {
-  state: 'ready';
+/** Process has started but is not yet accepting HTTP requests. */
+interface WaitingState {
+  state: 'waiting';
+  process: Child;
+}
+/** Process is ready to serve. */
+interface ServingState {
+  state: 'serving';
   process: Child;
 }
-type ServerState = StartingState | ReadyState;
+type ServerState = StartingState | WaitingState | ServingState;
 
 // A null value means that no one has requested a server yet.
 let serverProcess: ServerState | null = null;
 
-async function startServer() {
+async function startServer(servingCallback: () => void) {
   console.log('Starting RefStudio server');
   const command = new TauriCommand('call-sidecar', ['serve']);
   command.stderr.addListener('data', (line) => {
     console.log('server stderr', line);
+    if (line.includes('running on http')) {
```

Comment:
This matches:

    Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)

Definitely open to other ideas on how to tell that the app is listening for connections!